### ISSUE TITLE
Global policy does not throw an error

### ISSUE THREAD

**Issue by ms-eztut** [strapi/strapi#7133]
**Describe the bug**
Policies have no impact on custom graphql mutations.

**Steps to reproduce the behavior**

1. Create a new global policy with the cli
`module.exports = async(ctx, next) => ctx.unauthorized();`
2. Create a custom mutation in your **schema.grapgql.js**
3. Add the new policy to the mutation
```
 mutation: '
  createChild(input: createUserInput): Child
  ',
resolver: {
    Mutation: {
createChild: {
        description: 'Create a new child + user',
        policies: ['global::my-policy'],
        resolver: 'application::child.child.createChild',
      }}}
```
4. Add the function to the child controller
`module.exports.createChild = async (ctx) => console.log('controller called')`
5. The controller is called although the policy should return an error

**Expected behavior**
when a policy fails, the controller action shouldnt be called

**System**

- Node.js version: 12.18.2
- NPM version: 6.14.5
- Strapi version: 3.0.6
- Database: Mongo
- Operating system: Windows

**Additional context**
strapi is executed with the debugger in development mode:
`node --inspect-brk ./node_modules/strapi/bin/strapi.js dev`

**Comment by derrickmehaffy** [comment]
Hi @ms-eztut can you change the policy code to:
`module.exports = async(ctx, next) => return ctx.unauthorized();`
And see if that properly throws the error?

**Comment by ms-eztut** [comment]
Hey @derrickmehaffy, i've inserted your code and got the following error: 
`Error while running command develop: undefined`

Then i've modified your code to this:
`module.exports = async (ctx) => ctx.unauthorized();`

aand the request goes right through the restriction and still be executed.

**Comment by derrickmehaffy** [comment]
@ms-eztut are you on a mac?

**Comment by ms-eztut** [comment]
@derrickmehaffy  does this issue also exist on linux?

**Comment by derrickmehaffy** [comment]
> @derrickmehaffy does this issue also exist on linux?

I have tested on my own personal linux machine (Linux Mint 19.3), my Mac (Catalina), and my Windows machine (build 2004) and was not able to reproduce it, the last 2 users I saw with this issue were on macs. But I also have not seen this on any of the linux servers I have deployed nor tested on.

This is the same type of policy that I use in my [personal project](https://github.com/canonn-science/CAPIv2-Strapi/blob/a0c73077cac138dce7e6a4bc47ca14e983704ac9/config/policies/block-client.js#L23) as well.
